lib/response/headers.ex

```diff
@@ -9,6 +12,53 @@ defmodule HTTPServer.Response.Headers do
           allow: list(String.t())
         }
 
+  def build(
+        _req = %Request{method: "OPTIONS", path: path, headers: headers},
```

Comment:
```suggestion
        %Request{method: "OPTIONS", path: path, headers: headers},
```

Reply:
Is this OPTIONS match even needed? This function looks identical to the one bellow.

Reply:
I know, but I wasn't sure how I could solve for the fact that the status codes for an options request will be 200. I could deal with this in the response like:
```
def build(req = %Request{method: "OPTIONS"}, status_code, res_body, media_type) do
    %__MODULE__{}
    |> resource()
    |> status(status_code)
    |> body(res_body)
    |> headers(%{req | status_code: 405}, media_type)
    |> send_resp()
  end
```

or

```
def build(req, 405, res_body, media_type) do
    %__MODULE__{}
    |> resource()
    |> status(status_code)
    |> body(res_body)
    |> headers(%{req | method: "OPTIONS"}, media_type)
    |> send_resp()
  end
```
But this doesn't feel like it's necessarily sound logic?

Reply:
I see the difference now, thanks for pointing it out.